src/test/storage/encoded_segment_test.cpp

```diff
@@ -128,6 +128,33 @@ INSTANTIATE_TEST_CASE_P(
                       SegmentEncodingSpec{EncodingType::RunLength}, SegmentEncodingSpec{EncodingType::LZ4}),
     formatter);
 
+TEST_P(EncodedSegmentTest, SequentiallyReadEmptyIntSegment) {
+  const auto encoding_spec = GetParam();
+
+  // 1522
+  if (encoding_spec.encoding_type == EncodingType::RunLength ||
+      (encoding_spec.vector_compression_type.has_value() &&
+       *encoding_spec.vector_compression_type == VectorCompressionType::SimdBp128)) {
+    GTEST_SKIP();
+  }
+
+  auto value_segment = std::make_shared<ValueSegment<int32_t>>(pmr_concurrent_vector<int32_t>{});
+  auto base_encoded_segment = this->encode_value_segment(DataType::Int, value_segment);
+
+  EXPECT_EQ(value_segment->size(), base_encoded_segment->size());
+
+  resolve_encoded_segment_type<int32_t>(*base_encoded_segment, [&](const auto& encoded_segment) {
```

Comment:
any reason to leave these lines here if "nothing happens here"? If so, please doc, if no, please remove

Reply:
No since just the compression is tested. I'll remove them.

Reply:
As it turns out, this part causes the SIMDBP128 encoded segments to crash. Trying to iterate over the segment causes that. I added a comment.